packages/actor-function-factory-equality/lib/Equality.ts

```diff
@@ -0,0 +1,104 @@
+import { RegularFunction } from '@comunica/bus-function-factory';
+import { KeysExpressionEvaluator } from '@comunica/context-entries';
+import type {
+  DurationLiteral,
+  BooleanLiteral,
+
+  TimeLiteral,
+  LangStringLiteral,
+  Quad,
+} from '@comunica/expression-evaluator';
+import {
+  RegularOperator,
+  TypeAlias,
+  TypeURL,
+  bool,
+  declare,
+  dayTimeDurationsToSeconds,
+  defaultedDateTimeRepresentation,
+  defaultedDayTimeDurationRepresentation,
+  defaultedYearMonthDurationRepresentation,
+  toUTCDate,
+  yearMonthDurationsToMonths,
+  RDFEqualTypeError,
+} from '@comunica/expression-evaluator';
+
+/**
+ * https://www.w3.org/TR/sparql11-query/#func-RDFterm-equal
+ */
+export class Equality extends RegularFunction {
+  protected arity = 2;
+
+  public operator = RegularOperator.EQUAL;
```

Comment:
Currently, each  operat

Reply:
>  Not necessarily in this PR, but we need a plan, which this PR needs to be compatible with.

I agree. I also agree with the union of string. But I also think we should no longer differentiate between regular and special operators.

Reply:
> But I also think we should no longer differentiate between regular and special operators.

If that's feasible, that's ok for me!

Reply:
I think it should be feasable. This whole PR a a breaking change so I don't think it'll matter :))